src/main/java/it/pagopa/pn/bff/service/PublicKeysPgService.java

```diff
@@ -0,0 +1,217 @@
+                status,
+                xPagopaPnCxGroups
+        ).onErrorMap(WebClientResponseException.class, pnBffExceptionUtility::wrapException);
+    }
+
+    /**
+     * Rotate the public key identified by its id
+     *
+     * @param xPagopaPnUid        User Identifier
+     * @param xPagopaPnCxType     Customer/Receiver Type
+     * @param xPagopaPnCxId       Customer/Receiver Identifier
+     * @param xPagopaPnCxRole     User role
+     * @param id                  Unique identifier for the public key to be rotated
+     * @param bffPublicKeyRequest The publicKeyRequest parameter
+     * @param xPagopaPnCxGroups   Customer Groups
+     * @return BffPublicKeyResponse
+     */
+    public Mono<BffPublicKeyResponse> rotatePublicKey(String xPagopaPnUid, CxTypeAuthFleet xPagopaPnCxType,
+                                                      String xPagopaPnCxId, String xPagopaPnCxRole,
+                                                      String id, Mono<BffPublicKeyRequest> bffPublicKeyRequest,
+                                                      List<String> xPagopaPnCxGroups) {
+        log.info("Rotate public key {} status - senderId: {} - type: {} - groups: {}", id, xPagopaPnCxId, xPagopaPnCxType, xPagopaPnCxGroups);
+
+        return bffPublicKeyRequest.flatMap(request -> {
+            Mono<PublicKeyResponse> publicKeyResponse = pnPublickeyManagerClientPG.rotatePublicKey(
+                    xPagopaPnUid,
+                    CxTypeMapper.cxTypeMapper.convertPublicKeysPGCXType(xPagopaPnCxType),
+                    xPagopaPnCxId,
+                    xPagopaPnCxRole,
+                    id,
+                    PublicKeyRequestMapper.modelMapper.mapPublicKeyRequest(request),
+                    xPagopaPnCxGroups
+            ).onErrorMap(WebClientResponseException.class, pnBffExceptionUtility::wrapException);
+
+            return publicKeyResponse.map(PublicKeyResponseMapper.modelMapper::mapPublicKeyResponse);
+        });
+    }
+
+    /**
+     * Get the issuer status
+     *
+     * @param xPagopaPnUid      User Identifier
+     * @param xPagopaPnCxType   PG Type
+     * @param xPagopaPnCxId     PG id
+     * @return BffPublicKeysIssuerResponse
+     */
+    public Mono<BffPublicKeysCheckIssuerResponse> checkIssuerPublicKey(String xPagopaPnUid, CxTypeAuthFleet xPagopaPnCxType,
+                                                                       String xPagopaPnCxId) {
+
+        log.info("Public keys check issuer - senderId: {} - type: {}", xPagopaPnCxId, xPagopaPnCxType);
+
+        Mono<PublicKeysIssuerResponse> publicKeysIssuerResponse = pnPublickeyManagerClientPG.getIssuerStatus(
+                xPagopaPnUid,
+                CxTypeMapper.cxTypeMapper.convertPublicKeysPGCXType(xPagopaPnCxType),
+                xPagopaPnCxId
+        ).onErrorMap(WebClientResponseException.class, pnBffExceptionUtility::wrapException)
+        .map(PublicKeysIssuerStatusMapper.modelMapper::mapPublicKeysIssuerStatus);
```

Comment:
removing the mapper, the PublicKeysIssuerResponse import must be changed from the bff one to the micro-service one